internal/pkg/earpointer/analysis.go

```diff
@@ -64,7 +67,8 @@ func run(pass *analysis.Pass) (interface{}, error) {
 }
 
 // Insert into the state all the local references and global references.
-func (vis *visitor) initReferences(ssainput *buildssa.SSA) {
+// Argument "fns" stores the functions and methods found during this process.
```

Comment:
Why carry fns around as an output parameter? This is just all methods contained in ssainput.Pkg. I would do this after  of time in a run or generate them ahead of time. This keeps the functions simpler. If you need to accumulate this, you can add a field to visitor.